config/routes.rb

```diff
@@ -24,5 +26,17 @@
   get '/login' => 'sessions#login'
   get '/signup' => 'sessions#signup'
 
+  Sidekiq::Web.use Rack::Auth::Basic do |username, password|
+    # Protect against timing attacks:
+    # - See https://codahale.com/a-lesson-in-timing-attacks/
+    # - See https://thisdata.com/blog/timing-attacks-against-string-comparison/
+    # - Use & (do not use &&) so that it doesn't short circuit.
+    # - Use digests to stop length information leaking (see also ActiveSupport::SecurityUtils.variable_size_secure_compare)
+    ActiveSupport::SecurityUtils.secure_compare(::Digest::SHA256.hexdigest(username), ::Digest::SHA256.hexdigest(ENV["SIDEKIQ_USERNAME"])) &
```

Comment:
@castrolem as a better approach, we can use something similar to what Discourse does.

https://github.com/discourse/discourse/blob/888e68a1637ca784a7bf51a6bbb524dcf7413b13/lib/admin_constraint.rb
https://github.com/discourse/discourse/blob/master/config/routes.rb#L34